rclpy/src/rclpy/_rclpy.c

```diff
@@ -1418,6 +1455,47 @@ rclpy_send_response(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Check if a service server is available
+/**
+ * Raises ValueError if the arguments are not capsules
+ *
+ * \param[in] pynode Capsule pointing to the node the entity belongs to
+ * \param[in] pyclient Capsule pointing to the client
+ * \return True if the service server is available
+ */
+static PyObject *
+rclpy_service_server_is_available(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pynode;
+  PyObject * pyclient;
+
+  if (!PyArg_ParseTuple(args, "OO", &pynode, &pyclient)) {
+    return NULL;
+  }
+  if (!PyCapsule_IsValid(pynode, NULL) || !PyCapsule_IsValid(pyclient, NULL)) {
+    PyErr_Format(PyExc_ValueError, "Expected two PyCapsule as arguments");
```

Comment:
This raises ValueError on invalid capsule while rclpy_get_graph_guard_condition raises TypeError

Reply:
Now they both check if `PyCapsule_GetPointer` returns NULL, which means they raise ValueError 1628cbaecc1f30b429c47096c56975f577a579dd. Pull request #129 does the same thing. This will need to be updated to use capsule names in whichever PR lands second.